server/service/integration_android_software_test.go

```diff
@@ -496,3 +494,149 @@ func (s *integrationMDMTestSuite) enableAndroidMDM(t *testing.T) string {
+		batchAssociateAppStoreAppsRequest{
+			DryRun: false,
+			Apps: []fleet.VPPBatchPayload{
+				{AppStoreID: "app_1", SelfService: true, Platform: fleet.AndroidPlatform, Configuration: json.RawMessage("{}")},
+				{AppStoreID: "app_2", SelfService: true, Platform: fleet.AndroidPlatform, Configuration: json.RawMessage("{}")},
+				{AppStoreID: "app_3", SelfService: true, Platform: fleet.AndroidPlatform, Configuration: json.RawMessage("{}")},
+				{AppStoreID: "app_4", SelfService: true, Platform: fleet.AndroidPlatform, Configuration: json.RawMessage("{}")},
+			},
+		}, http.StatusOK, &batchResp,
+	)
+
+	s.DoJSON("GET", "/api/latest/fleet/software/titles", nil, http.StatusOK, &listSWTitles, "team_id", fmt.Sprint(0))
+	s.Assert().Len(listSWTitles.SoftwareTitles, 4)
+
+	// Update configurations
+	s.DoJSON("POST", "/api/latest/fleet/software/app_store_apps/batch",
+		batchAssociateAppStoreAppsRequest{
+			DryRun: false,
+			Apps: []fleet.VPPBatchPayload{
+				{AppStoreID: "app_1", Platform: fleet.AndroidPlatform, Configuration: nil},
+				{AppStoreID: "app_2", Platform: fleet.AndroidPlatform, Configuration: exampleConfiguration},
+				{AppStoreID: "app_3", Platform: fleet.AndroidPlatform, Configuration: json.RawMessage("{}")},
+				{AppStoreID: "app_4", Platform: fleet.AndroidPlatform, Configuration: json.RawMessage("{}")},
+			},
+		},
+		http.StatusOK, &batchResp, "team_name", t.Name(),
+	)
+
+	s.DoJSON("GET", "/api/latest/fleet/software/titles", nil, http.StatusOK, &listSWTitles, "team_id", fmt.Sprint(*teamID))
+	s.Assert().Len(listSWTitles.SoftwareTitles, 4)
+
+	var titleResp getSoftwareTitleResponse
+	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/software/titles/%d", titleApp1), &getSoftwareTitleRequest{
+		ID:     titleApp1,
+		TeamID: teamID,
+	}, http.StatusOK, &titleResp)
+	require.Equal(t, "app_1", *titleResp.SoftwareTitle.ApplicationID)
+	require.Equal(t, json.RawMessage(`{}`), titleResp.SoftwareTitle.AppStoreApp.Configuration)
+
+	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/software/titles/%d", titleApp2), &getSoftwareTitleRequest{
+		ID:     titleApp2,
+		TeamID: teamID,
+	}, http.StatusOK, &titleResp)
+	require.Equal(t, "app_2", *titleResp.SoftwareTitle.ApplicationID)
+	require.Contains(t, string(titleResp.SoftwareTitle.AppStoreApp.Configuration), "workProfileWidgets")
+
+	// Remove 2 other apps, 2 configurations should remain
+	s.DoJSON("POST", "/api/latest/fleet/software/app_store_apps/batch",
+		batchAssociateAppStoreAppsRequest{
+			DryRun: false,
+			Apps: []fleet.VPPBatchPayload{
+				{AppStoreID: "app_1", Platform: fleet.AndroidPlatform, Configuration: nil},
+				{AppStoreID: "app_2", Platform: fleet.AndroidPlatform, Configuration: exampleConfiguration},
+			},
+		},
+		http.StatusOK, &batchResp, "team_name", t.Name(),
+	)
+
+	s.DoJSON("GET", "/api/latest/fleet/software/titles", nil, http.StatusOK, &listSWTitles, "team_id", fmt.Sprint(*teamID))
+	s.Assert().Len(listSWTitles.SoftwareTitles, 2)
```

Comment:
should we add a check here to make sure that `app_1` has an empty config and `app_2` has `exampleConfiguration`?